site/content/agent-app-widgets/js-api.md

```diff
@@ -101,7 +101,191 @@ It appends given message at the end of current conversation input window or into
+  3. Go to the `Building blocks` section of your app and create a new Agent App widget. You need to provide the `Widget source URL`, which can be either an URL pointing to the Web (for eg. `https://livechat.github.io/sample-widget-integration/`) or to localhost (for eg. `https://localhost:4000`), if you want to use the local version of the widget.
+  4. Set the `Widget placement` to `Chat details` and provide the initial state JSON, eg.: 
+
+```json
+{
+  "customSections": [
+    { 
+      "title": "Example section",
+      "components": [
+        {
+          "type": "button",
+          "data": {
+            "label": "Example button",
+            "id": "example-button"
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+  6. The sections list can only be initialized once and they can't be modified afterwarts. This means you can't add or remove a section using the SDK later on. You can however modify the list of components inside a section later on.
+  7. Go to `Private installation` and click the `Install app` button.
+  8. Your widget should be now visible in the right sidebar of the Agent App chat view. Beware - if you did not set any icon for your widget, it may appear that the widget not there. To make sure, you can hover over the widget icon bar and check if there's any clickable whitespace. You can set the icon in Developers Console.
+
+### How to update a section
+
+To update a section you can use the `modifyCustomerDetailsSection` method like this:
+
+```js
+LiveChat.modifyCustomerDetailsSection(
+  { 
+    "title": "Example section",
+    "components": [
+      {
+        "type": "button",
+        "data": {
+          "label": "Updated button",
+          "id": "example-button"
+        }
+      }
+    ]
+  } 
+);
+```
+---
+**IMPORTANT**
+
+The **title** attribute in the root of the JSON acts as an ID to the section and *cannot* be changed. You need to provide it to identify which section you want to update.
+
+---
+
+### Component types
+Components are *lego bricks* which can be used for building a section.
+
+#### Title
+Title could be used in several cases. Component look depends on given data.
+
+| Property | Required | Type | Note |
+|----------|----------| ---- | ----------- |
+| `title` | Yes | string |                   |
+| `value` | No | string |                    |
+| `description` | No | string |              |
+| `imgUrl` | No | string |                   |
+| `imgSize` | No | string, one of: ["small", "big"]` | Default value: "big" |
+| `link` | No | string | URL added when title is a link. |
+
+> Example of title component
+
+```json
+{
+    type: 'title',
+    data: {
+        title: 'title',
+        value: 'value',
+        description: 'description',
+        imgUrl: 'https://www.gstatic.com/webp/gallery/4.jpg'
+    }
+}
+```
+
+#### Button
+Simple button from design system
```

Comment:
Be careful, properties listed in the table don't match the JSON.

Also, you should mention that you can listen for button clicks using the SDK:
```javascript
Livechat.on("customer_details_section_button_click", ({ buttonId }) => {
  console.log(buttonId);
});
```

The `buttonId` will be the `id` specified in the schema, so if you want to capture a specific click, you need to make sure that the `id` is unique across all definitions.

Reply:
Fixed, I'll also check the other ones after I fix the issues you listed.